### DISCUSSION TITLE
How to manage multiple websites based on a single template repo?

### DISCUSSION THREAD

**Discussion by gxjansen** [community/community#128165] [New to GitHub]
I have a general git(hub) question. Fairly new to using git (at least for the purpose below) and Googling (or ChatGPTing) doesn't really give me answers (or rather: way to many different answers). I think I need more specific search terms but I don't know which so maybe you can give me a hint :smile:

I'm looking for info on how to work with Git when you would want to set up a "base" repository (which in itself is a fork but that's probably not relevant) and use that as a template/base repo for other projects (websites in my case). I am the sole admin of all these websites and I want them all to share (most of) the layout/design/theme (that should come from the base repo) but each site will have different content, in a different language and will have some minor differences like (like URL, name, site logo, meta description, etc.).

My main challenge is: How do I set up a system that (manually or automatically) updates the 4 websites when the base theme is updated, but prevent it from overwriting any customizations?

- Should I start with a base repo for the "main template" and then create 4 forks for the different websites?
- Or do I use a single repository with multiple branches, one for each website?
- Do I need to figure out Git submodules/subtrees?
- Should I use GitHub’s template repository feature?
- Or would the preferred way be to work with a regular git merge base/upstream and use .gitattributes to ignore certain files/folders that are specific to the 4 sites?
- Or shouldn't I be using git(hub) for this to begin with…?

In case it makes a difference: I plan on creating these websites with [Astro](https://astro.build/) and publish through GitHub Pages.

**Comment by ebndev** [comment]
Hi @gxjansen, thanks for being a part of the GitHub Community!

If there will only be minor differences in the content of the site, you can have a json or env file that you can customize. When you update the template, you can then pull changes without overwriting those files. (Like your idea about using .gitattributes to ignore site specific files).

I personally would create separate forks so the branches don't get mixed up when deploying to pages or if you want to make more changes later.

If you need more complex modifications for each site, I do see that Astro has [astro:env](https://astro.build/blog/astro-4100/) as a feature now that might help you manage this better!
